implementations/rust/ockam/ockam_transport_tcp/src/transport/portals.rs

```diff
@@ -143,3 +146,67 @@ impl TcpTransport {
         Ok(())
     }
 }
+
+/// Result of [`TcpTransport::create_inlet`] call.
+#[derive(Clone, Debug)]
+pub struct TcpInlet {
+    socket_address: SocketAddr,
+    processor_address: Address,
+    outlet_listener_route: Arc<RwLock<Route>>,
+}
+
+impl fmt::Display for TcpInlet {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Socket: {}, Processor: {}",
+            self.socket_address, self.processor_address
+        )
+    }
+}
+
+impl TcpInlet {
+    /// Constructor
+    pub fn new(
+        socket_address: SocketAddr,
+        processor_address: Address,
+        outlet_listener_route: Arc<RwLock<Route>>,
+    ) -> Self {
+        Self {
+            socket_address,
+            processor_address,
+            outlet_listener_route,
+        }
+    }
+
+    /// Socket Address
+    pub fn socket_address(&self) -> SocketAddr {
+        self.socket_address
+    }
+
+    /// Processor address
+    pub fn processor_address(&self) -> &Address {
+        &self.processor_address
+    }
+
+    /// Update the route to the outlet node.
+    ///  NOTE: Existing TCP connections will still use the old route,
+    ///        only newly accepted connections will use the new route.
+    pub fn update_outlet_node_route(&self, new_route: Route) -> Result<()> {
```

Comment:
It would be good to explain in the comment who would use this method, and when it would be used.